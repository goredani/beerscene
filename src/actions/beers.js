import uuid from 'uuid';



// ADD BEER

export const addBeer = (
    {   
        beername ='',
        beertype = '',
        brewery = '',
        ABV ='', 
        IBU='', 
        origin = '',
        description = '', 
        price='',
        rating = '',
        createdAt=0
    } = {}
) => ({
    type: 'ADD_BEER',
    beer: {
        id: uuid(),
        beername,
        beertype,
        brewery,
        ABV,
        IBU,
        origin,
        description,
        price,
        rating,
        createdAt
    }
});

//REMOVE

export const removeBeer = ({ id } = {}) => ({
    type: 'REMOVE_BEER',
    id
});

//EDIT BEER

export const editBeer = ( id, updates) => ({
    type: 'EDIT_BEER',
    id,
    updates
});