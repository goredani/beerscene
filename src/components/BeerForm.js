import React from 'react';
import moment from  'moment';


let regex = /^\d{1,}(\.\d{0,2})?$/;

export default class BeerForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            beername: props.beer ? props.beer.beername : '',
            beertype: props.beer ? props.beer.beertype : '',
            brewery: props.beer ? props.beer.brewery : '',
            description: props.beer ? props.beer.description : '',
            price: props.beer ? (props.beer.price / 100).toString() : '',
            rating: props.beer ? (props.beer.rating / 100).toString() : '',
            ABV: props.beer ? (props.beer.ABV / 100).toString() : '',
            IBU: props.beer ? (props.beer.IBU / 100).toString() : '',
            origin: props.beer ? props.beer.origin : '',
            createdAt: '',
            error: ''
        };
    }

    onBeerNameChange = (e) => {
        const beername = e.target.value;
        this.setState(() => ({beername}));
    };

    onBeerTypeChange = (e) => {
        const beertype = e.target.value;
        this.setState(() => ({beertype}));
    };

    onBreweryChange = (e) => {
        const brewery = e.target.value;
        this.setState(() => ({brewery}));
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({description}));
    };
    onABVChange = (e) => {
        const ABV = e.target.value;

        if (!ABV || ABV.match(regex)) {
            this.setState(() => ({ ABV }))
        }
    };
    onIBUChange = (e) => {
        const IBU = e.target.value;
        if (!IBU || IBU.match(regex)) {
            this.setState(() => ({IBU}))
        }
    };
    onOriginChange = (e) => {
        const origin = e.target.value;
        this.setState(() => ({ origin }));
    };
 
    onPriceChange = (e) => {
        const price = e.target.value;
        if (!price || price.match(regex)) {
            this.setState(() => ({price}));
        }
    };

    onRatingChange = (e) => {
        const rating = e.target.value;
        if (!rating || rating.match(regex)) {
            this.setState(() => ({rating}));
        }
    };

    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({createdAt}));
        }

    }

    

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.beername) {
            this.setState(() => ({error: 'Please provide the name of the beer'}));
        } else if (!this.state.beertype) {
            this.setState(() => ({error: 'Please provide the type of the beer'}));
        } else {
            this.setState(()=> ({ error: ''}));
            this.props.onSubmit({
                beername: this.state.beername,
                beertype: this.state.beertype,
                brewery: this.state.brewery,
                ABV: parseFloat(this.state.ABV, 10 *100),
                IBU: parseFloat(this.state.IBU, 10 *100),
                brewery: this.state.origin,
                description: this.state.description,
                price: parseFloat(this.state.price, 10) *100,
                rating: parseFloat(this.state.rating, 10) *100,
                createdAt: moment().format('YYYY-MM-DD HH:MM')
            });
        }
    };
    
    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Beer Name"
                        autoFocus
                        value={this.state.beername}
                        onChange={this.onBeerNameChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder="Beer Type"
                        value = {this.state.beertype}
                        onChange = {this.onBeerTypeChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder="Brewery"
                        value = {this.state.brewery}
                        onChange = {this.onBreweryChange}
                    />
                    <br />
                    <input
                        type="number"
                        placeholder="ABV"
                        value={this.state.ABV}
                        onChange={this.onABVChange}
                    />
                    <br />
                    <input
                        type="number"
                        placeholder="IBU"
                        value={this.state.IBU}
                        onChange={this.onIBUChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder="Country Origin"
                        value={this.state.origin}
                        onChange={this.onOriginChange}
                    />
                    <br />
                    <textarea
                        placeholder="Description"
                        value={this.state.description}
                        onChange={this.onDescriptionChange}>
                    </textarea>
                    <br />
                    <input
                        type="number"
                        placeholder="Price"
                        value={this.state.price}
                        onChange={this.onPriceChange}
                    />
                    <br />
                    <input
                        type="number"
                        placeholder="Rating"
                        value={this.state.rating}
                        onChange={this.onRatingChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder={moment().format('YYYY, ddd, hA')}
                     />
                  
                    
                   
                    <button>Add Beer</button>
                </form>
            </div>
        )
    }
}

