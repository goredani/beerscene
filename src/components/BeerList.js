import React from 'react';
import { connect } from 'react-redux';
import BeerListItem from './BeerListItem';
import selectBeers from '../selectors/beers';

export const BeerList = (props) => (
    <div>
        <h1> Beer List </h1> 
        {
            props.beers.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.beers.map((beer) => {
                    return <BeerListItem key={beer.id} {...beer}/>
                })
            )

        }
    
    </div>
);

const mapStateToProps = (state) => {
    return {
        beers: selectBeers(state.beers, state.filters)
    };
};

export default connect(mapStateToProps)(BeerList);

