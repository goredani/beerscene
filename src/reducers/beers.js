

const beersReducerDefaultState = [];

export default (state = beersReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_BEER':
            return [
                ...state,
                action.beer
            ];
        case 'REMOVE_BEER':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_BEER':
            return state.map((beer) => {
                if(beer.id === action.id) {
                    return {
                        ...beer,
                        ...action.updates
                    };
                } else {
                    return beer;
                }
            }); 
        default:
            return state;
    }
};

