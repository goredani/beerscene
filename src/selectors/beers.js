export default (beers, { text, sortBy, startDate, endDate }) => {
    return beers.filter((beer) => {
      const startDateMatch = typeof startDate !== 'number' || beer.createdAt >= startDate;
      const endDateMatch = typeof endDate !== 'number' || beer.createdAt <= endDate;
      const textMatch = beer.beername.toLowerCase().includes(text.toLowerCase());
  
      return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if (sortBy === 'rating') {
            return a.rating > b.rating ? 1 : -1;
        } 
        else if (sortBy === 'beername') {
            return a.beername > b.beername ? 1 : -1;
        } 
        else if (sortBy === 'beertype') {
            return a.beertype > b.beertype ? 1 : -1;
        } 
        else if (sortBy === 'brewery') {
            return a.brewery > b.brewery ? 1 : -1;
        } 
        else if (sortBy === 'ABV') {
            return a.ABV > b.ABV ? 1 : -1;
        } 
        else if (sortBy === 'IBU') {
            return a.IBU > b.IBU ? 1 : -1;
        }
        else if (sortBy === 'price') {
            return a.price > b.price ? 1 : -1;
        }
    });
  };