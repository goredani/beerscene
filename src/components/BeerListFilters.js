import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByABV} from '../actions/filters';


const BeerListFilters = (props) => (
    <div>
        <input 
            type='text' 
            value={props.filters.text} 
            onChange={(e) => {
                props.dispatch(setTextFilter(e.target.value));
            }} 
        />
        <p>Sort By</p>
        <select
            value={props.filters.sortBy}
            onChange={(e) => {
                if (e.target.value === 'date') {
                    props.dispatch(sortByDate());
                } else if (e.target.value === 'ABV') {
                    props.dispatch(sortByABV());
                } 
            }}
        >
            <option value="date">Date</option>
            <option value="ABV">ABV</option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(BeerListFilters);