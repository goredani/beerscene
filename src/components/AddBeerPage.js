import React from 'react';
import { connect } from 'react-redux';
import BeerForm from './BeerForm';
import { addBeer } from '../actions/beers';

export class AddBeerPage extends React.Component {
  onSubmit = (beer) => {
    this.props.addBeer(beer);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Beer</h1>
        <BeerForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addBeer: (beer) => dispatch(addBeer(beer))
});


export default connect(undefined, mapDispatchToProps)(AddBeerPage);
