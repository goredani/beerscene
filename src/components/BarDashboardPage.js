import React from 'react';
import BeerList from './BeerList';
import BeerListFilters from './BeerListFilters';

const BarDashboardPage = () => (
  <div>
    <BeerListFilters/>
    <BeerList />
  </div>
);

export default BarDashboardPage;
