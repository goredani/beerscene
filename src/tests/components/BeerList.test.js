import React from 'react';
import { shallow } from 'enzyme';
import { BeerList } from '../../components/BeerList';
import beers from '../fixtures/beers'; 

test('should render BeerList with beers', () => {
   const wrapper = shallow(<BeerList beers={beers}/>);
   expect(wrapper).toMatchSnapshot(); 
});

test('should render BeerList with empty message', () => {
    const wrapper = shallow(<BeerList beers={beers}/>);
    expect(wrapper).toMatchSnapshot();
});