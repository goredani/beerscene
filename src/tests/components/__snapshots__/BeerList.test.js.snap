// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BeerList with beers 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeerList
    beers={
      Array [
        Object {
          "ABV": 4.8,
          "IBU": 1,
          "beername": "Babbaba",
          "beertype": "Hefeweizen",
          "brewery": "Primátor",
          "createdAt": 0,
          "description": "Bababab",
          "id": "1",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        Object {
          "ABV": 4.8,
          "IBU": 2,
          "beername": "Rererere",
          "beertype": "Hefeweizen",
          "brewery": "Csecs",
          "createdAt": -345600000,
          "description": "Fefefe",
          "id": "2",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        Object {
          "ABV": 4.8,
          "IBU": 3,
          "beername": "Nyenye",
          "beertype": "Hefeweizen",
          "brewery": "Cseceeeelkfjs",
          "createdAt": 345600000,
          "description": "TEtete",
          "id": "3",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           Beer List 
        </h1>,
        Array [
          <BeerListItem
            ABV={4.8}
            IBU={1}
            beername="Babbaba"
            beertype="Hefeweizen"
            brewery="Primátor"
            createdAt={0}
            description="Bababab"
            id="1"
            origin="Czech Republic"
            price={195}
            rating={3.56}
          />,
          <BeerListItem
            ABV={4.8}
            IBU={2}
            beername="Rererere"
            beertype="Hefeweizen"
            brewery="Csecs"
            createdAt={-345600000}
            description="Fefefe"
            id="2"
            origin="Czech Republic"
            price={195}
            rating={3.56}
          />,
          <BeerListItem
            ABV={4.8}
            IBU={3}
            beername="Nyenye"
            beertype="Hefeweizen"
            brewery="Cseceeeelkfjs"
            createdAt={345600000}
            description="TEtete"
            id="3"
            origin="Czech Republic"
            price={195}
            rating={3.56}
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Beer List ",
        },
        "ref": null,
        "rendered": " Beer List ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "ABV": 4.8,
          "IBU": 1,
          "beername": "Babbaba",
          "beertype": "Hefeweizen",
          "brewery": "Primátor",
          "createdAt": 0,
          "description": "Bababab",
          "id": "1",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "ABV": 4.8,
          "IBU": 2,
          "beername": "Rererere",
          "beertype": "Hefeweizen",
          "brewery": "Csecs",
          "createdAt": -345600000,
          "description": "Fefefe",
          "id": "2",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "ABV": 4.8,
          "IBU": 3,
          "beername": "Nyenye",
          "beertype": "Hefeweizen",
          "brewery": "Cseceeeelkfjs",
          "createdAt": 345600000,
          "description": "TEtete",
          "id": "3",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             Beer List 
          </h1>,
          Array [
            <BeerListItem
              ABV={4.8}
              IBU={1}
              beername="Babbaba"
              beertype="Hefeweizen"
              brewery="Primátor"
              createdAt={0}
              description="Bababab"
              id="1"
              origin="Czech Republic"
              price={195}
              rating={3.56}
            />,
            <BeerListItem
              ABV={4.8}
              IBU={2}
              beername="Rererere"
              beertype="Hefeweizen"
              brewery="Csecs"
              createdAt={-345600000}
              description="Fefefe"
              id="2"
              origin="Czech Republic"
              price={195}
              rating={3.56}
            />,
            <BeerListItem
              ABV={4.8}
              IBU={3}
              beername="Nyenye"
              beertype="Hefeweizen"
              brewery="Cseceeeelkfjs"
              createdAt={345600000}
              description="TEtete"
              id="3"
              origin="Czech Republic"
              price={195}
              rating={3.56}
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Beer List ",
          },
          "ref": null,
          "rendered": " Beer List ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "ABV": 4.8,
            "IBU": 1,
            "beername": "Babbaba",
            "beertype": "Hefeweizen",
            "brewery": "Primátor",
            "createdAt": 0,
            "description": "Bababab",
            "id": "1",
            "origin": "Czech Republic",
            "price": 195,
            "rating": 3.56,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "ABV": 4.8,
            "IBU": 2,
            "beername": "Rererere",
            "beertype": "Hefeweizen",
            "brewery": "Csecs",
            "createdAt": -345600000,
            "description": "Fefefe",
            "id": "2",
            "origin": "Czech Republic",
            "price": 195,
            "rating": 3.56,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "ABV": 4.8,
            "IBU": 3,
            "beername": "Nyenye",
            "beertype": "Hefeweizen",
            "brewery": "Cseceeeelkfjs",
            "createdAt": 345600000,
            "description": "TEtete",
            "id": "3",
            "origin": "Czech Republic",
            "price": 195,
            "rating": 3.56,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render BeerList with empty message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeerList
    beers={
      Array [
        Object {
          "ABV": 4.8,
          "IBU": 1,
          "beername": "Babbaba",
          "beertype": "Hefeweizen",
          "brewery": "Primátor",
          "createdAt": 0,
          "description": "Bababab",
          "id": "1",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        Object {
          "ABV": 4.8,
          "IBU": 2,
          "beername": "Rererere",
          "beertype": "Hefeweizen",
          "brewery": "Csecs",
          "createdAt": -345600000,
          "description": "Fefefe",
          "id": "2",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        Object {
          "ABV": 4.8,
          "IBU": 3,
          "beername": "Nyenye",
          "beertype": "Hefeweizen",
          "brewery": "Cseceeeelkfjs",
          "createdAt": 345600000,
          "description": "TEtete",
          "id": "3",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           Beer List 
        </h1>,
        Array [
          <BeerListItem
            ABV={4.8}
            IBU={1}
            beername="Babbaba"
            beertype="Hefeweizen"
            brewery="Primátor"
            createdAt={0}
            description="Bababab"
            id="1"
            origin="Czech Republic"
            price={195}
            rating={3.56}
          />,
          <BeerListItem
            ABV={4.8}
            IBU={2}
            beername="Rererere"
            beertype="Hefeweizen"
            brewery="Csecs"
            createdAt={-345600000}
            description="Fefefe"
            id="2"
            origin="Czech Republic"
            price={195}
            rating={3.56}
          />,
          <BeerListItem
            ABV={4.8}
            IBU={3}
            beername="Nyenye"
            beertype="Hefeweizen"
            brewery="Cseceeeelkfjs"
            createdAt={345600000}
            description="TEtete"
            id="3"
            origin="Czech Republic"
            price={195}
            rating={3.56}
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Beer List ",
        },
        "ref": null,
        "rendered": " Beer List ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "ABV": 4.8,
          "IBU": 1,
          "beername": "Babbaba",
          "beertype": "Hefeweizen",
          "brewery": "Primátor",
          "createdAt": 0,
          "description": "Bababab",
          "id": "1",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "ABV": 4.8,
          "IBU": 2,
          "beername": "Rererere",
          "beertype": "Hefeweizen",
          "brewery": "Csecs",
          "createdAt": -345600000,
          "description": "Fefefe",
          "id": "2",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "ABV": 4.8,
          "IBU": 3,
          "beername": "Nyenye",
          "beertype": "Hefeweizen",
          "brewery": "Cseceeeelkfjs",
          "createdAt": 345600000,
          "description": "TEtete",
          "id": "3",
          "origin": "Czech Republic",
          "price": 195,
          "rating": 3.56,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             Beer List 
          </h1>,
          Array [
            <BeerListItem
              ABV={4.8}
              IBU={1}
              beername="Babbaba"
              beertype="Hefeweizen"
              brewery="Primátor"
              createdAt={0}
              description="Bababab"
              id="1"
              origin="Czech Republic"
              price={195}
              rating={3.56}
            />,
            <BeerListItem
              ABV={4.8}
              IBU={2}
              beername="Rererere"
              beertype="Hefeweizen"
              brewery="Csecs"
              createdAt={-345600000}
              description="Fefefe"
              id="2"
              origin="Czech Republic"
              price={195}
              rating={3.56}
            />,
            <BeerListItem
              ABV={4.8}
              IBU={3}
              beername="Nyenye"
              beertype="Hefeweizen"
              brewery="Cseceeeelkfjs"
              createdAt={345600000}
              description="TEtete"
              id="3"
              origin="Czech Republic"
              price={195}
              rating={3.56}
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Beer List ",
          },
          "ref": null,
          "rendered": " Beer List ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "ABV": 4.8,
            "IBU": 1,
            "beername": "Babbaba",
            "beertype": "Hefeweizen",
            "brewery": "Primátor",
            "createdAt": 0,
            "description": "Bababab",
            "id": "1",
            "origin": "Czech Republic",
            "price": 195,
            "rating": 3.56,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "ABV": 4.8,
            "IBU": 2,
            "beername": "Rererere",
            "beertype": "Hefeweizen",
            "brewery": "Csecs",
            "createdAt": -345600000,
            "description": "Fefefe",
            "id": "2",
            "origin": "Czech Republic",
            "price": 195,
            "rating": 3.56,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "ABV": 4.8,
            "IBU": 3,
            "beername": "Nyenye",
            "beertype": "Hefeweizen",
            "brewery": "Cseceeeelkfjs",
            "createdAt": 345600000,
            "description": "TEtete",
            "id": "3",
            "origin": "Czech Republic",
            "price": 195,
            "rating": 3.56,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
